options/m_option.c

```diff
@@ -2594,15 +2594,52 @@ static char *print_rel_time(const m_option_t *opt, const void *val)
     case REL_TIME_PERCENT:
         return talloc_asprintf(NULL, "%g%%", t->pos);
     }
-    return talloc_strdup(NULL, "none");
+    return talloc_strdup(NULL, "no");
+}
+
+static char *pretty_print_rel_time(const m_option_t *opt, const void *val)
+{
+    const struct m_rel_time *t = val;
+    switch(t->type) {
+    case REL_TIME_ABSOLUTE:
+        return mp_format_time(t->pos, false);
+    }
+
+    return print_rel_time(opt, val);
+}
+
+static int rel_time_set(const m_option_t *opt, void *dst,
+                        struct mpv_node *src)
+{
+    struct m_rel_time *t = dst;
+    int rc = double_set(opt, &t->pos, src);
+    if (rc > 0)
+        t->type = REL_TIME_ABSOLUTE;
+    return rc;
+}
+
+static int rel_time_get(const m_option_t *opt, void *ta_parent,
+                        struct mpv_node *dst, void *src)
+{
+    struct m_rel_time *t = src;
+    if (t->type == REL_TIME_ABSOLUTE)
+        return double_get(opt, ta_parent, dst, &t->pos);
+
+    dst->format = MPV_FORMAT_STRING;
+    dst->u.string = print_rel_time(opt, t);
+    ta_set_parent(dst->u.string, ta_parent);
+    return 1;
 }
 
 const m_option_type_t m_option_type_rel_time = {
     .name  = "Relative time or percent position",
     .size  = sizeof(struct m_rel_time),
     .parse = parse_rel_time,
     .print = print_rel_time,
+    .pretty_print = pretty_print_rel_time,
     .copy  = copy_opt,
```

Comment:
This is also missing .add compared to `m_option_type_time`, shall I check if the code could want to call it?

Reply:
pretty_print is strictly meant for OSD only and can change arbitrarily, so there's no need to check anything.

Reply:
.add is used for M_PROPERTY_SWITCH (i.e., cycle add input commands). Not sure what if anything uses these for time. The first thing that comes to mine is stuff like sub-delay, but that's actually a DOUBLE, so wouldn't be affected.